tests/actions/scan_test.py

```diff
@@ -106,6 +106,15 @@ def test_does_not_write_state_when_dry_run(self, mock_file_operations):
 
         assert not mock_file_operations.write.called
 
+    def test_does_not_write_state_when_scan_head(self, mock_file_operations):
+        with self.setup_env(
+            SecretsCollection(),
+            '--scan-head',
+        ) as args:
+            assert scan_repo(args) == 0
+
+        assert not mock_file_operations.write.called
+
```

Comment:
Can we write a test that this `--scan-head` functionality actually works (rather than just not saving state)?